test/deprecations.test.ts

```diff
@@ -0,0 +1,11 @@
+import * as fs from 'node:fs'
+import * as path from 'node:path'
+import {test, expect} from 'vitest'
+import {tsFileErrors} from './ts-output'
+
+test.each(['usage.test.ts', 'types.test.ts'])('%s: toMatchTypeOf matches toExtend behaviour', file => {
```

Comment:
Minor nit: can we actually use [`test.for`](https://vitest.dev/api/#test-for) with a `TestContext`?